v3/app/models/user.rb

```diff
@@ -0,0 +1,114 @@
+class User < ActiveRecord::Base
+
+    has_secure_password
+
+    def self.validate_token(options, check_role_id = 0) # check_role_id default to 0 for 'any'
```

Comment:
All of this token stuff does not seem like it belongs in the user model

Reply:
While I agree about the login/logout being in the user controller, this code is about validation, not identity.  I would put it in a separate class that makes it clear it is doing user validation.

Reply:
Let's use "token_controller" and something like "token/create" and "token/delete"

Reply:
DONE.